File: src/BaseHmiTypes/Screens/Widgets/HmiThreshold.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public enum HmiThresholdValueMode
{
    Absolute,
    Percentage
}

public sealed class HmiThreshold
{
    public int? Index { get; set; }

    public HmiProperty<double>? Value { get; set; }

    public HmiProperty<HmiColor>? Color { get; set; }

    public HmiProperty<bool>? Blink { get; set; }

    public HmiProperty<HmiColor>? EndColor { get; set; }

    public HmiProperty<double>? GradientStop { get; set; }

    public string? GradientAxis { get; set; }

    public HmiGradientDirection? GradientDirection { get; set; }
}

File: typescript/base-hmi-types/src/screens/widgets/HmiThreshold.ts
import { HmiColor, HmiGradientDirection, HmiProperty } from "../base.js";

export enum HmiThresholdValueMode {
  Absolute = "Absolute",
  Percentage = "Percentage",
}

export class HmiThreshold {
  index?: number;
  value?: HmiProperty<number>;
  color?: HmiProperty<HmiColor>;
  blink?: HmiProperty<boolean>;
  endColor?: HmiProperty<HmiColor>;
  gradientStop?: HmiProperty<number>;
  gradientAxis?: string;
  gradientDirection?: HmiGradientDirection;
}
